Quest system

NickRisse opened the pull request:
The quest system is working and ready to be pushed to main.

Please review the system as described in the Jira task.

NickRisse committed:
Revert "Increased Base Animation Speed"

This reverts commit 10a8758486f6e30de43599886984050a9abcca76.

NickRisse committed:
Confusion

I don't know what I commit anymore.

NickRisse committed:
Renewed Quest system

Started working on the renewed quest system. Created needed folders, started with the first scripts, prefabs and scriptable objects.

NickRisse committed:
Add layout to gitignore

NickRisse committed:
Continued working on the quest system

Added a game events manager, quest events, and a quest point script.
Current issue: I need to understand how our input manager works so that I can reference the submit button and more.

NickRisse committed:
Finished base code + base quest structure

I added the remaining base code for the quest system and added a base set up for the quests. These quests are sorted in multiple folders and have various different quest steps.

TO-DO:
- Figure out the input manager to make the submit button work as part of the QuestPoint.cs script to enable and disable the different quest steps and trigger points.
- Some quest step scrips need to be filled with code entirely.

All the way at the end: Set up all quests and rewards properly and add the last bits of code.

NickRisse committed:
Trying to make the input work

Adding input changes

MGHMSJ5 committed:
Trying out some fixing

fix(quest): changed the quest system scene by deactivating all except 1 npc for testing purpose. Changed the BaseInteract and QuestPoint script to try to fix the error.
(It only needs testing if it now works)

NickRisse committed:
Trying to make quests work

MGHMSJ5 committed:
Fixed error regarding quest step

fix(quest): changed the scene, investigation prefab and the quest step script.
The issue was that the ID was 0, so (comparing to the QuestPoint script), the id is now set

MGHMSJ5 committed:
Removed the change

MGHMSJ5 committed:
Fixed quest spawn position

fix(quest spawn): removed the parent transform in the initiating of the quest

NickRisse committed:
Continued working on the Quest system

Added
- RewardEvents.cs

Changed
- NPCInteractEXAMPLE.prefab
- Quest System
-> All Quests
- GameEventsManager.cs
- QuestInfoSO.cs
- QuestManager.cs
- QuestPoint.cs

Current Issue:
- For some reason I receive an error as soon as two or more QuestPoint scripts are in the scene. This should not happen as the start and end point of a quest could be different NPCs. Also one character could hold multiple QuestPoints as they might be the starting or end point for multiple quests.

MGHMSJ5 committed:
Error fix (hopefully)

fix(GameEventsManager null error): delayed the OnEnabled coe in QuestPoint script

NickRisse committed:
Trying to fix the error

The error persists that something is not set to an instance.

SirTangoGibus committed:
Quest System Fix

fix: Added GameEventsManager to project settings > Script  Execution Order.

NickRisse committed:
Final push

Everything should work now. One thing to change later is to remove the "needs to have NPC example" from the quest point.

NickRisse committed:
Merge branch 'main' into QuestSystem

MGHMSJ5 commented:
Hey @NickRisse, while reviewing your PR, I'd suggest the following code changes:

👉 [**Code Suggestion for #22**](https://gitkraken.dev/drafts/bf40c8c0-9b5b-43d8-8f52-f9c5ad0bf9ac?source=PRComment)
>I would suggest these changes for two scripts that were causing an error. This had to do with getting the current amount of screws. This script was changed a bit during the development of the Quest System

https://github.com/MGHMSJ5/Project-Capstone/pull/22

You can also review and apply these suggestions [locally on your machine](https://gitkraken.dev/link/drafts/bf40c8c0-9b5b-43d8-8f52-f9c5ad0bf9ac?type=suggested_pr_change&prEntityId=WyJnaXRodWIiLCJwciIsIjEiLCIiLCJQUl9rd0RPT1BsZ2I4NlYtNUdjIl0=&source=PRComment).

<details><summary>Learn more about GitKraken Code Suggest</summary>

>[Code Suggest](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbS9zb2x1dGlvbnMvY29kZS1zdWdnZXN0P3NvdXJjZT1naXRodWIuY29tJm1lZGl1bT1wcl9jb21tZW50?source=PRSuggest&event=redirectToCodeSuggestPage&provider=github) liberates your code reviews from GitHub's restrictive, comment-only feedback style. As simple as suggesting changes in a Google-doc, provide real code suggestions from where you code, e.g. your IDE, and on anything in your project — not just on the lines of code changed in the PR.

_Join your team on [GitKraken](https://gitkraken.dev/redirect/aHR0cHM6Ly93d3cuZ2l0a3Jha2VuLmNvbT9zb3VyY2U9Z2l0aHViLmNvbSZtZWRpdW09cHJfY29tbWVudA==?source=PRSuggest&event=redirectToIndexPage&provider=github) to speed up PR review._
</details>

NickRisse committed:
Collect Resources changes

The code for quest steps in which the player needs to collect screws has been adjusted.

NickRisse committed:
Quest and new NPC Interact

MGHMSJ5 committed:
Dialogue manager in quest system scene

fix(quest-dialogue): put in all the references for the dialogue manager in the quest system scene

MGHMSJ5 reviewed (approved):
i approve ☺
